modules/claims_api/lib/bd/bd.rb

```diff
@@ -22,7 +22,15 @@ def search(claim_id, file_number)
       body = { data: { claimId: claim_id, fileNumber: file_number } }
       ClaimsApi::Logger.log('benefits_documents',
                             detail: "calling benefits documents search for claimId #{claim_id}")
-      client.post('documents/search', body)&.body&.deep_symbolize_keys
+      res = client.post('documents/search', body)&.body
+
+      raise ::Common::Exceptions::GatewayTimeout.new(detail: 'Upstream service error.') unless res.is_a?(Hash)
+
+      res.deep_symbolize_keys
+    rescue ::Common::Exceptions::GatewayTimeout => e
+      ClaimsApi::Logger.log('benefits_documents',
+                            detail: "/search failure for claimId #{claim_id}, #{e.message}")
+      raise
     rescue => e
       ClaimsApi::Logger.log('benefits_documents',
                             detail: "/search failure for claimId #{claim_id}, #{e.message}")
```

Comment:
Do we want to re-raise all exceptions as we do in `upload`? i.e., why return an empty object here but not in `upload`? If we re-raise all exceptions, we could remove the above `rescue` block.

Reply:
re-raising slightly works, but we also have differing standards now... which we can't really break in the currently "in production" services (Claims Status) but we can introduce moving forward, so the goal is to abide to [Lighthouse Standards](https://department-of-veterans-affairs.github.io/va-api-standards/errors/), which have slightly different format than the common exceptions did (maybe not for all cases, but the general assumption is that they have slightly differing formats).

Since we rescue and return an empty object in the controller I think we're fine to introduce a "Lighthouse Standard" Gateway error here.

Reply:
on that note, we probably want silent failures here for gateway timeouts too. it still allows Veterans access to their claim information, it would just return an empty list of supporting documents, so we would provide _the data we have_.

Reply:
Makes sense - will remove this added `rescue` block and use the existing, general `rescue` (thereby logging the gateway timeout and returning an empty object to the controller).

Reply:
https://github.com/department-of-veterans-affairs/vets-api/pull/17389/commits/71c1f3760968e43d65b309b909988158635d79de